File: java/src/main/java/exercice/fibonacci/Fibonacci.java
package exercice.fibonacci;

/**
 * @author llin
 * @created 14/05/14.
 */
public class Fibonacci {
    public static int fibonacci(int nb) {
        if (nb <= 1) {
            return 1;
        }

        int result = 1;
        int prevResult = 1;
        for (int i = 2; i <= nb; i++) {
            int tmp = result;
            result += prevResult;
            prevResult = tmp;
        }
        return result;
    }

    public static int fibonacciRecursive(int nb) {
        if (nb <= 1) {
            return 1;
        }

        return fibonacci(nb - 1) + fibonacci(nb - 2);
    }
}

File: golang/exercice/fibonacci/fibonacci.go
package fibonacci

func Fibonacci(a int) int {
	if a == 1 || a == 2 {
		return 1
	}
	return Fibonacci(a - 1) + Fibonacci(a - 2)
}

func FibonacciArr(a int) []int {
	b := make([]int, a)
	b[0], b[1] = 1, 1

	for i := 2; i < a; i++ {
		b[i] = b[i - 1] + b[i - 2]
	}
	return b
}
